aptos-move/block-executor/src/executor.rs

```diff
@@ -1000,13 +1116,24 @@ where
     }
 
     fn apply_output_sequential(
-        unsync_map: &UnsyncMap<T::Key, T::Value, X, T::Identifier>,
+        unsync_map: &UnsyncMap<T::Key, T::Tag, T::Value, X, T::Identifier>,
         output: &E::Output,
-    ) {
+    ) -> ::std::result::Result<(), PanicOr<IntentionalFallbackToSequential>> {
         for (key, (write_op, layout)) in output.resource_write_set().into_iter() {
             unsync_map.write(key, write_op, layout);
         }
 
+        // TODO[agg_v2](fix): provide layouts
+        for (group_key, _, group_ops) in output.resource_group_write_set().into_iter() {
```

Comment:
@vusirikala, if we need to get the metadata for reads in sequential execution as well, here we can store metadata_op in unsync_map (as a normal write) and then access that (similar to how parallel execution stores metadata_op as a normal resource write at group_key)